### ISSUE TITLE
Asterisk indicating required field is not announced by JAWS

### ISSUE THREAD

**Issue by mgifford** [ckeditor/ckeditor-dev#2500]
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1.    Create a formatted text field and make it required
2.    Open JAWS v18.0.2945.
3.    Create or edit a node that has that field
4.    Use the Tab key to navigate to the field

### Expected result

All asterisks and the meaning of the asterisks are expected to be announced by JAWS.

### Actual result

Description: When a CKeditor Field is is required and has the required Asterisk, JAWS does not announce that the field is required.

Example: Within the “Create Article” page, the “Body*” field is announced by JAWS as, “Rich text editor body field type in text”. Instead, JAWS is expected to announce the required field as e.g. “Rich text editor body required field, type in text”.

-    Testing was performed using JAWS v18.0.2945.
-    This defect exists in IE11 and Google Chrome v68.0.3440.106.

## Other details

Originally reported here https://www.drupal.org/project/drupal/issues/2998629

* Browser: IE11 and Google Chrome v68.0.3440.106.
* OS: Windiows

**Comment by mlewand** [comment]
We have no such API as of yet, but it make sense to provide a way to mark CKE4 instance/field being required. For instance assume that it's simply a bool `config.required`.

As far as implementation goes, I have checked this issue and adding a `[aria-required=true]` to a `[role=textbox]` element does the job as you can see in [my CKEditor using aria-required attribute (#2500) codepen](https://codepen.io/mlewand/pen/vVQzdm).

However there's one more dependency, for some reason classic editor does not have `[role=textbox]` specified, it needs to be checked what was the reason for it. `aria-required` **works even with classic** (on JAWS) but that's screen reader quirk rather than reliable solution, as per spec there are handful of roles that do support this attribute.

**Comment by mgifford** [comment]
Thanks @mlewand 

I'll try to get a response back to you.
